main.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    (0...length).each do |i|
+      yield self[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    (0...length).each { |i| yield self[i], i }
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    r = []
+    my_each { |x| r << x if yield x }
+    r
+  end
+
+  def my_all?(pattern = nil)
+    r = true
+    if block_given?
+      my_each { |x| r = false unless yield x }
+    elsif pattern
+      my_each { |x| r = false unless pattern === x }
+    else
+      my_each { |x| r = false unless x }
+    end
+    r
+  end
+
+  def my_any?(pattern = nil)
+    r = false
+    if block_given?
+      my_each { |x| r = true if yield x }
+    elsif pattern
+      my_each { |x| r = true if pattern === x }
+    else
+      my_each { |x| r = true if x }
+    end
+    r
+  end
+
+  def my_none?(pat = nil)
+    if block_given?
+      my_each { |x| return false if yield(x) }
+    elsif !pat.nil?
+      if pat.is_a?(Class)
+        my_each { |x| return false if x.is_a?(pat) }
+      elsif pat.is_a?(Regexp)
+        my_each { |x| return false if pat.match(x.to_s) }
+      else
+        my_each { |x| return false if x == pat }
+      end
+    else
+      my_each do |x|
+        return false if x
+      end
+    end
+    true
+  end
+
+  def my_count(number = nil)
+    count = 0
+    if !block_given?
+      count = length
+    elsif !number.nil?
+      my_each { |x| count += 1 if x == number }
+    else
+      my_each { |x| count += 1 if yield(x) }
+    end
+    count
+  end
```

Comment:
[REQUIRED]

Counts method works not correct.

        Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq 
          array.count(LOWEST_VALUE)
     
                expected: 11
                     got: 100